src/storage/mutate/UpdateEdgeProcessor.cpp

```diff
@@ -313,18 +313,22 @@ std::string UpdateEdgeProcessor::updateAndWriteBack(PartitionID partId,
 }
 
 
-bool UpdateEdgeProcessor::checkFilter(const PartitionID partId,
+UpdateEdgeProcessor::FilterResult UpdateEdgeProcessor::checkFilter(const PartitionID partId,
                                       const cpp2::EdgeKey& edgeKey) {
     auto ret = collectEdgesProps(partId, edgeKey);
-    if (ret != kvstore::ResultCode::SUCCEEDED) {
-        return false;
+    if (insertable_ && ret == kvstore::ResultCode::ERR_KEY_NOT_FOUND) {
```

Comment:
If insertable_ is true,  we should return succeeded if key not found.  Otherwise, we should return failed.